File: ports/java/src/core/containers/BcsVector.java
package bcs.core.containers;

import java.util.ArrayList;

// BcsVector is mapped to Java's native ArrayList with framework-specific semantics
public class BcsVector<T> {
    private final ArrayList<T> items;

    public BcsVector() {
        this.items = new ArrayList<>();
    }

    public void append(T item) {
        this.items.add(item);
    }

    public T at(int index) {
        return this.items.get(index);
    }

    public int size() {
        return this.items.size();
    }

    public void clear() {
        this.items.clear();
    }
}

File: ports/csharp/src/core/containers/BcsVector.cs
using System.Collections.Generic;

namespace BCS.Core.Containers {
    // BcsVector is mapped to C#'s native List<T> with framework-specific semantics
    public class BcsVector<T> {
        private List<T> _items;

        public BcsVector() {
            _items = new List<T>();
        }

        public void Append(T item) {
            _items.Add(item);
        }

        public T At(int index) {
            return _items[index];
        }

        public int Size() {
            return _items.Count;
        }

        public void Clear() {
            _items.Clear();
        }
    }
}
